src/features/applicationView/similarCustomersTable/__tests__/utils.test.ts

```diff
@@ -0,0 +1,49 @@
+import { getFilteredCustomersData } from '../utils';
+import { mockFilteredCustomers } from '../__fixtures__/mockData';
+import { FILTERED_CUSTOMERS } from '../__generated__/FILTERED_CUSTOMERS';
+
+describe('utils', () => {
+  describe('getFilteredCustomersData', () => {
+    it('should map filtered customers data', () => {
+      expect(getFilteredCustomersData(mockFilteredCustomers)).toMatchSnapshot();
+    });
+
+    it('should return an empty array when there are no customers', () => {
+      const testData: FILTERED_CUSTOMERS = {
+        profiles: {
+          __typename: 'ProfileNodeConnection',
+          count: 0,
+          edges: [],
+        },
+      };
+      expect(getFilteredCustomersData(testData)).toEqual([]);
+    });
+
+    it('should remove the edges with a value of null', () => {
```

Comment:
This can actually never happen, the basic problem is that the GraphQL contract is wrong from this point of view. So it might be waste to create a test for it.

Reply:
And this comment applies to all places we deal with edges and nodes.

Reply:
If this can indeed _never_ happen, should we instruct TypeScript with `as`? For example `(edge as PROFILE_EDGE).node as PROFILE_NODE`.

Leaving the `if (!edge?.node) return acc;` would leave the untested branch noise into the test coverage table.

Reply:
Another impossibility I found is `t` function returning `undefined` in Headers of `SimilarCustomersTable`, which I handled with `as`. It would require an invalid localization config to occur, and the app would blow up elsewhere before reaching that point

Reply:
Great idea, let's try the `as`